feat: quick button additions for main app page and profile page

BrandonDwyer opened the pull request:
Added a few lines to be able to logout from the main app page and profile page, and also view the profile page from the main app page

BrandonDwyer committed:
feat: added landing page for app

BrandonDwyer committed:
style: removed classnames and updated page content

BrandonDwyer committed:
feat: quick button additions for main app page and profile page

BrandonDwyer committed:
style: removed unused export line

omarkmu commented on src/pages/App.js:
```diff
@@ -65,6 +67,12 @@ export default function App() {
         {foods.map((elem) => <li>{elem.description}</li>)}
       </p>
+      <Row>
+        <AnchorButton href="/app" text="Profile" />
```
I think the `href` here is meant to be `"/profile"`.

BrandonDwyer replied:
Yes thanks for catching this.

omarkmu committed:
fix: Fixed bug in required field validation

The extra attribute should only be added if the value is empty.

omarkmu committed:
Merge pull request #39 from omarkmu/fix/input-required-validation

fix: Fixed bug in required field validation

BrandonDwyer committed:
fix: fixed home page not immediatly loading

omarkmu committed:
chore: Added React Router

omarkmu committed:
chore: Modified AnchorButton to use Router logic
chore: Changed all anchor tags to Link components

omarkmu committed:
style: Removed outdated TODO comments

BrandonDwyer committed:
fix: fixed href in main app and changed button positions

omarkmu committed:
Merge pull request #40 from omarkmu/chore/set-up-react-router

Set up React Router

BrandonDwyer committed:
chore: removed logout and main page hyperlinks to focus this commit solely on landing page

omarkmu commented on src/pages/App.js:
```diff
@@ -38,7 +40,13 @@ export default function App() {
-      <h1> Find Out Best Overall Diets</h1>
+      <h1> BotTrition</h1>
+      <Row>
+        <AnchorButton href="/profile" text="View Profile" />
```
The React Router addition changed the `href` prop in `AnchorButton` to `to`. Changing `href` to `to` should be the last change necessary to ensure this works with the newest changes. It may be a good idea to pull from main as well, to ensure the test cases pass.

BrandonDwyer replied:
I am going to update all AnchorButton hrefs -> to and then commit again to verify that this was done properly.

omarkmu committed:
test: Add tests for profile page

omarkmu committed:
chore: Added hidden label for inches select
fix: Changed to react-style select default values

Changed the Select component to silence some React warnings.

BrandonDwyer committed:
Merge pull request #42 from omarkmu/test/add-profile-tests

Add profile tests

BrandonDwyer committed:
fix: updated button routing to handle new react routing capability

BrandonDwyer committed:
fix: fixed main app page last href changing href -> to

BrandonDwyer commented:
This PR should be good to merge into the main branch with another quick review.

BrandonDwyer committed:
fix: minor changes to resolve merge conflicts

BrandonDwyer committed:
Merge branch 'main' into feat/landing-page

BrandonDwyer committed:
Merge pull request #37 from omarkmu/feat/landing-page

feat: added landing page for app

BrandonDwyer committed:
fix: installed react router to be able to pass test case before merging into main

BrandonDwyer committed:
fix: fixed react pages not redirecting

BrandonDwyer committed:
fix: fixed converting href -> to and resolving logout blank page error by redirecting to login page

BrandonDwyer commented:
I fixed the branch and it is ready to be pushed locally however I have run into an issue, and I am not sure how to go about this.

Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

omarkmu commented:
Which git commands did you run to pull from main? I don't think that should be the result of a normal git pull. Either way, trying git pull as suggested in the error might help.

BrandonDwyer committed:
fix: resolved merge conflicts

BrandonDwyer committed:
fix: resolved app test not passing
